include/revng/Pipes/RootKind.h

```diff
@@ -4,15 +4,20 @@
 // This file is distributed under the MIT License. See LICENSE.md for details.
 //
 
+#include "llvm/ADT/STLExtras.h"
+
 #include "revng/Pipeline/Context.h"
 #include "revng/Pipeline/GenericLLVMPipe.h"
 #include "revng/Pipeline/Kind.h"
 #include "revng/Pipeline/LLVMContainer.h"
 #include "revng/Pipeline/LLVMGlobalKindBase.h"
+#include "revng/Pipes/ModelInvalidationRule.h"
+#include "revng/TupleTree/TupleTreeDiff.h"
+#include "revng/TupleTree/TupleTreePath.h"
 
 namespace revng::pipes {
 
-class RootKind : public pipeline::LLVMKind {
+class RootKind : public pipeline::LLVMKind, ModelInvalidationRule {
```

Comment:
I see that you want to compose things and make things optional, but all this flexibility makes it really really hard to explain what's going on.

When I had to explain to @pfez how to create a `Kind`, he wanted to dig under the surface (rightly so) and if you start from `TaggedFunctionKind`, it's quite a journey. We have 1) kinds that can or cannot enumerate containers, 2) we have a decoupling for LLVM, 3) kinds that are coupled to `llvm::Function`s through `FunctionTags` and I might still be missing something.
Clearly 3) is really needed, but the others to me keep striking as complexity with little to no return.

tl;dr: I'm happy to see that you opted for having `Kind::verify` to be mandatory, can we do the same for `getInvalidations` and avoid further increasing complexity?